pkg/daemon/ceph/osd/remove.go

```diff
@@ -58,8 +59,25 @@ func RemoveOSDs(context *clusterd.Context, clusterInfo *client.ClusterInfo, osds
 		if status == upStatus {
 			logger.Infof("osd.%d is healthy. It cannot be removed unless it is 'down'", osdID)
 			continue
+		} else {
+			logger.Infof("osd.%d is marked 'DOWN'", osdID)
+		}
+
+		// Check we can remove the OSD in case jobs are fired in parallel
+		// loop forever until the osd is ok-to-stop (the job might timeout eventually but it's
+		// better than loosing the osd data...)
+		for {
+			_, err := client.OSDOkToStop(context, clusterInfo, osdID, 1)
+			if err != nil {
```

Comment:
What about also checking for ok-to-destroy? In that case, a force flag would really be necessary since there may not yet be an OSD where the PGs can move.